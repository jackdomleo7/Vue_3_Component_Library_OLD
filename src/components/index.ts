/* eslint-disable import/prefer-default-export */
export { default as JSample } from './Sample/Sample.vue'
