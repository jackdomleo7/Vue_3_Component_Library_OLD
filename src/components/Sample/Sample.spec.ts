import { mount, shallowMount } from '@vue/test-utils'
import { JSample } from '@/components'
import { configureAxe, toHaveNoViolations } from 'jest-axe'

expect.extend(toHaveNoViolations)

const axe = configureAxe({
  rules: {
    region: { enabled: false } // Don't need to check because components are loaded in isolation
  }
})

describe('Sample.vue', () => {
  /**
   * Vue Instance
   */

  it('is a Vue instance', () => {
    const wrapper = shallowMount(JSample)

    expect(wrapper.vm).toBeTruthy()
  })

  /**
   * Snapshots
   */

  it('matches snapshot when mounted', () => {
    const wrapper = mount(JSample)

    expect(wrapper.element).toMatchSnapshot()
  })

  it('matches snapshot when shallow mounted', () => {
    const wrapper = shallowMount(JSample)

    expect(wrapper.element).toMatchSnapshot()
  })

  /**
   * A11y
   */

  it('to have no obvious accessibility violations', async () => {
    const wrapper = mount(JSample)

    expect(await axe(wrapper.element)).toHaveNoViolations()
  })
})
