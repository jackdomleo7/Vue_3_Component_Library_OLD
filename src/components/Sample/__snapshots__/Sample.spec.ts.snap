// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sample.vue matches snapshot when mounted 1`] = `
<div
  class="sample"
>
  <p>
    The counter was initialized to 
    <b>
      5
    </b>
    .
  </p>
  <button>
     Click +1 
  </button>
  <button>
     Click -1 
  </button>
  <button>
     Click +5 
  </button>
  <button>
     Click -5 
  </button>
  <button>
     Reset 
  </button>
</div>
`;

exports[`Sample.vue matches snapshot when shallow mounted 1`] = `
<div
  class="sample"
>
  <p>
    The counter was initialized to 
    <b>
      5
    </b>
    .
  </p>
  <button>
     Click +1 
  </button>
  <button>
     Click -1 
  </button>
  <button>
     Click +5 
  </button>
  <button>
     Click -5 
  </button>
  <button>
     Reset 
  </button>
</div>
`;

exports[`address-block.vue matches snapshot when mounted 1`] = `
<div
  class="sample"
>
  <p>
    The counter was initialized to 
    <b>
      5
    </b>
    .
  </p>
  <button>
     Click +1 
  </button>
  <button>
     Click -1 
  </button>
  <button>
     Click +5 
  </button>
  <button>
     Click -5 
  </button>
  <button>
     Reset 
  </button>
</div>
`;

exports[`address-block.vue matches snapshot when shallow mounted 1`] = `
<div
  class="sample"
>
  <p>
    The counter was initialized to 
    <b>
      5
    </b>
    .
  </p>
  <button>
     Click +1 
  </button>
  <button>
     Click -1 
  </button>
  <button>
     Click +5 
  </button>
  <button>
     Click -5 
  </button>
  <button>
     Reset 
  </button>
</div>
`;
